DatadogSessionReplay/SRSnapshotTests/SRSnapshotTests/Utils/SnapshotTestCase.swift

```diff
@@ -36,23 +38,24 @@ internal class SnapshotTestCase: XCTestCase {
         let scheduler = TestScheduler()
         let processor = Processor(queue: NoQueue(), writer: Writer())
         let recorder = try Recorder(
-            configuration: configuration,
+            processor: processor
+        )
+        let recordingCoordinator = RecordingCoordinator(
+            scheduler: scheduler,
+            privacy: configuration.privacy,
             rumContextObserver: RUMContextObserverMock(),
-            processor: processor,
-            scheduler: scheduler
+            srContextPublisher: SRContextPublisher(core: PassthroughCoreMock()),
+            recorder: recorder,
+            sampler: Sampler(samplingRate: 1000)
         )
-
         // Set up wireframes interception and trigger recorder once:
         var wireframes: [SRWireframe]?
 
         processor.interceptWireframes = {
             wireframes = $0
             expectation.fulfill()
         }
-
-        recorder.start()
         scheduler.triggerOnce()
-        recorder.stop()
```

Comment:
suggestion/ With `recorder.captureNextRecord()` being now internal, we can do much simpler:
```swift
// Set up wireframes interception and trigger recorder once:
var wireframes: [SRWireframe]?

processor.interceptWireframes = {
    wireframes = $0
    expectation.fulfill()
}
recorder.captureNextRecord(
    .init(privacy: configuration.privacy, applicationID: "", sessionID: "", viewID: "", viewServerTimeOffset: nil)
)
```
this way getting rid of `TestScheduler` and `RUMContextObserverMock`. Also this will solve the compiler warning:
```
Initialization of immutable value 'recordingCoordinator' was never used; consider replacing with assignment to '_' or removing it
```
Less dependencies here == less maintenance and broken builds 🙂.